File: app/src/main/java/com/fivesysdev/standwithukraine/mvp/Contract.java
package com.fivesysdev.standwithukraine.mvp;

import com.fivesysdev.standwithukraine.data.DayStatistic;

import java.time.LocalDate;

public interface Contract {
    interface View {
        void setDayStatistic(DayStatistic dayStatistic);
        void blockingNextButton(boolean isBlocked);
    }
    interface Model {
        DayStatistic getNextDayStatistic();
        DayStatistic getPreviousDayStatistic();
        DayStatistic getCurrentDayStatistic();
        LocalDate getDate();
        void setDate(String date);

    }
    interface Presenter {
        void onPreviousButtonClick();
        void onNextButtonClick();
        DayStatistic getDayStatistic();
        void onDestroy();
        void setDate(String date);
    }
}

File: app/src/main/java/com/fivesysdev/standwithukraine/mvp/Contract.kt
package com.fivesysdev.standwithukraine.mvp

import com.fivesysdev.standwithukraine.data.DayStatistic
import java.time.LocalDate

interface Contract {
    interface View {
        fun setDayStatistic(dayStatistic: DayStatistic?)
        fun blockingNextButton(isBlocked: Boolean)
    }

    interface Model {
        fun getNextDayStatistic(): DayStatistic?
        fun getPreviousDayStatistic(): DayStatistic?
        fun getCurrentDayStatistic(): DayStatistic?
        var date: LocalDate
    }

    interface Presenter {
        fun onPreviousButtonClick()
        fun onNextButtonClick()
        fun getDayStatistic(): DayStatistic?
        fun setDate(date: String?)
    }
}
